app/src/main/cpp/skyline/gpu/texture/layout.cpp

```diff
@@ -89,144 +89,428 @@ namespace skyline::gpu::texture {
+                );
+
+                gobStartY = 0;
+                // Only offset the pitch surface by the amount of lines we actually copied
+                pitchRob += pitchWidthBytes * (robHeight - startingRobLinePadding);
+            }
+        }
+
+        for (u16 rob{}; rob < subSurfaceHeightRobs; ++rob) {
+            deswizzleSubRob(pitchRob, std::false_type{});
+            pitchRob += robBytes; // Increment the pitch ROB to the next subROB
+        }
+
+        if (endsSubRectYMisaligned) {
+            const u16 endingGobLineOffset{(u16)(endingRobLineOffset % GobHeight)};
+            blockHeight = util::DivideCeil<u16>(endingRobLineOffset, GobHeight);
+
+            deswizzleSubRob(pitchRob,
+                            std::true_type{},
+                            0,
+                            0,
+                            (gobBlockHeight - blockHeight) * GobWidth * GobHeight,
+                            endingGobLineOffset ? endingGobLineOffset : GobHeight
             );
         }
     }
 
     void CopyBlockLinearToLinear(Dimensions dimensions, size_t formatBlockWidth, size_t formatBlockHeight, size_t formatBpb, size_t gobBlockHeight, size_t gobBlockDepth, u8 *blockLinear, u8 *linear) {
         CopyBlockLinearInternal<true>(
             dimensions,
-            formatBlockWidth, formatBlockHeight, formatBpb,
+            formatBlockWidth, formatBlockHeight, formatBpb, (u32)(dimensions.width * formatBpb),
             gobBlockHeight, gobBlockDepth,
             blockLinear, linear
         );
     }
 
+    void CopyBlockLinearToPitch(Dimensions dimensions,
+                                size_t formatBlockWidth, size_t formatBlockHeight, size_t formatBpb, u32 pitchAmount,
+                                size_t gobBlockHeight, size_t gobBlockDepth,
+                                u8 *blockLinear, u8 *pitch) {
+        CopyBlockLinearInternal<true>(
+                dimensions,
+                formatBlockWidth, formatBlockHeight, formatBpb, pitchAmount,
+                gobBlockHeight, gobBlockDepth,
+                blockLinear, pitch
+        );
+    }
+
+    void CopyBlockLinearToPitchSubrect(Dimensions pitchDimensions, Dimensions blockLinearDimensions,
+                                       size_t formatBlockWidth, size_t formatBlockHeight, size_t formatBpb, u32 pitchAmount,
+                                       size_t gobBlockHeight, size_t gobBlockDepth,
+                                       u8 *blockLinear, u8 *pitch,
+                                       u16 originX, u16 originY) {
+        CopyBlockLinearSubrectInternal<true>(pitchDimensions, blockLinearDimensions,
+                                             formatBlockWidth, formatBlockHeight, formatBpb, pitchAmount,
+                                             gobBlockHeight, gobBlockDepth,
+                                             blockLinear, pitch,
+                                             originX, originY);
```

Comment:
Run the formatter on this to fix all these weird indentations, also push off the closing bracket onto the next line for multi-line functions.